beamformer/unit_test/beamform_op_sequence_test.py

```diff
@@ -0,0 +1,214 @@
+        - polarisations (p) = 2, always
+        - n_blocks = 16, always
+        - samples_per_block calculated as n_samples_per_channel // n_blocks
+
+Contains one test (parametrised):
+    1. The first test uses the list of values present in test/test_parameters.py to run the
+        kernel through a range of value combinations.
+"""
+
+import numpy as np
+import pytest
+from beamforming import reorder as cpu_reorder
+from beamforming.beamform_op_sequence import OpSequenceTemplate
+from katsdpsigproc import accel
+from unit_test import complex_mult_cpu, test_parameters
+from unit_test.coeff_generator_cpu import CoeffGenerator
+
+
+@pytest.mark.parametrize("n_batches", test_parameters.n_batches)
+@pytest.mark.parametrize("n_ants", test_parameters.n_ants)
+@pytest.mark.parametrize("n_channels", test_parameters.n_channels)
+@pytest.mark.parametrize("n_samples_per_channel", test_parameters.n_samples_per_channel)
+@pytest.mark.parametrize("n_beams", test_parameters.n_beams)
+@pytest.mark.parametrize("xeng_id", test_parameters.xeng_id)
+@pytest.mark.parametrize("samples_delay", test_parameters.samples_delay)
+@pytest.mark.parametrize("phase", test_parameters.phase)
+def test_beamform_op_sequence(
+    n_batches: int,
+    n_ants: int,
+    n_channels: int,
+    n_samples_per_channel: int,
+    n_beams: int,
+    xeng_id: int,
+    samples_delay: int,
+    phase: float,
+):
+    """
+    Parametrised unit test of the beamform computation using Numba-based kernel.
+
+    This unit test runs the computation on a combination of parameters indicated in test_parameters.py. The values
+    parametrised are indicated in the parameter list, operating on a *single* batch. This unit test also invokes
+    verification of the beamformed data.
+
+    Parameters
+    ----------
+    n_batches:
+        Number of batches to process.
+    n_ants:
+        The number of antennas from which data will be received.
+    n_channels:
+        The number of frequency channels out of the FFT.
+        NB: This is not the number of FFT channels per stream.
+        The number of channels per stream is calculated from this value.
+    n_samples_per_channel:
+        The number of time samples per frequency channel.
+    n_beams:
+        The number of beams that will be steered.
+    xeng_id:
+        Identify of the XEngine. This is used to compute the actual channel numbers processed per engine.
+    samples_delay:
+        Delay in ADC samples that should be applied.
+    phase:
+        Phase value in radians to be applied.
+
+    This test:
+        1. Populate a host-side array with random data in the range of the relevant dtype.
+        2. Instantiate the pre-beamform reorder and pass data to it.
+        3. Instantiate the beamform coefficient generator.
+        2. Instantiate the beamformer complex multiplication.
+        3. Grab the output, beamformed data.
+        4. Verify it relative to the input array using a reference computed on the host CPU.
```

Comment:
Your concord isn't quite right here. I'd suggest the following:
```suggestion
    This test will:
        1. Populate a host-side array with random data in the range of the relevant dtype.
        2. Instantiate the pre-beamform reorder and pass data to it.
        3. Instantiate the beamform coefficient generator.
        2. Instantiate the beamformer complex multiplication.
        3. Grab the output, beamformed data.
        4. Verify it relative to the input array using a reference computed on the host CPU.
```